kairon/shared/actions/utils.py

```diff
@@ -501,6 +502,51 @@ def prepare_response(response_template: str, http_response: Any):
 
         return parsed_output
 
+    @staticmethod
+    def handle_utter_bot_response(dispatcher: CollectingDispatcher, dispatch_type: str, bot_response: Any):
+        from json import JSONDecodeError
+
+        message = None
+        if dispatch_type == DispatchType.json.value:
+            try:
+                if isinstance(bot_response, str):
+                    bot_response = json.loads(bot_response)
+                dispatcher.utter_message(json_message=bot_response)
+            except JSONDecodeError as e:
+                message = f'Failed to convert http response to json: {str(e)}'
+                logger.exception(e)
+                dispatcher.utter_message(bot_response)
+        else:
+            dispatcher.utter_message(bot_response)
+        return bot_response, message
+
+    @staticmethod
+    def filter_out_kairon_system_slots(slots: dict):
+        from kairon.shared.constants import KaironSystemSlots
+
+        slot_values = {slot: value for slot, value in slots.items() if
+                       slot not in {item.value for item in KaironSystemSlots}}
+        return slot_values
+
+    @staticmethod
+    def run_pyscript(source_code: Text, context: dict):
+        pyscript_evaluator_url = Utility.environment['evaluator']['pyscript']['url']
+        request_body = {"source_code": source_code, "predefined_objects": context}
+        if not ActionUtility.is_empty(pyscript_evaluator_url):
+            resp = ActionUtility.execute_http_request(pyscript_evaluator_url, "POST", request_body)
+            if resp.get('error_code') != 0:
+                raise ActionFailure(f'Pyscript evaluation failed: {resp}')
+            result = resp.get('data')
+        else:
+            lambda_response = CloudUtility.trigger_lambda(EventClass.pyscript_evaluator, request_body)
+            if lambda_response['StatusCode'] != 200:
+                raise ActionFailure(f"{lambda_response}")
+            result = lambda_response["Payload"].get('body')
+
+        response_type = DispatchType.json.value if isinstance(result, dict) else DispatchType.text.value
```

Comment:
remove this. lets not decide the type of utterance on user's behalf. 
just assume that the evaluator response is of type: {"bot_response": "some response", "slots": {"is_new_user": true}, "type": "text"}
If the slots key is not of type dict, the that must also be ignored